### DISCUSSION TITLE
Error in lesson 4

### DISCUSSION THREAD

**Discussion by arbdev95** [Cyfrin/foundry-full-course-cu#113] [Q&A] [answered]
This is the code:

contract FundMe {

    uint256 public minimumUsd = 5;

    function Fund() public payable {
          //allow users to send $
         //have a minimum $ sent
         //how do we send ETH to this contract? 
         require(msg.value > minimumUsd, "didn't sent enough ETH") ; //1 ETH=1e18=1000000000000000000 wei //the "require" keyword is a checker
          //revert=undo any actions that have been done, and send the remaining gas back   
        
          

         function getPrice() public{
             //Address: 0x694AA1769357215DE4FAC081bf1f309aDC325306
              // ABI: 
              AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306)

         }
          function getConversionRate() public {}                                                                       

    }

It continue to say as an error:

ParserError: Expected '(' but got identifier
  --> FundMe.sol:24:19:
   |
24 |          function getPrice() public{
   |                   ^^^^^^^^
Why? It seems everything correct

**Comment by palmcivet7** [comment, marked as answer]
Your `Fund()` function is not closed properly. Here is the corrected code:
```
contract FundMe {

    uint256 public minimumUsd = 5;

    function Fund() public payable {
        //allow users to send $
        //have a minimum $ sent
        //how do we send ETH to this contract? 
        require(msg.value > minimumUsd, "didn't sent enough ETH") ; //1 ETH=1e18=1000000000000000000 wei //the "require" keyword is a checker
        //revert=undo any actions that have been done, and send the remaining gas back   
    }

    function getPrice() public {
        //Address: 0x694AA1769357215DE4FAC081bf1f309aDC325306
        // ABI: 
        // AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306)
    }

    function getConversionRate() public {}  
}
```

**Reply by cromewar** [reply]
this is most likely the answer, please @arbdev95 let us know, does this worked for you?
